src/cc/libclient/KfsClient.cc

```diff
@@ -4943,9 +4943,47 @@ KfsClientImpl::UpdateFilesize(int fd)
 }
 
 void
-KfsClientImpl::GetLayout(GetLayoutOp& inOp)
+KfsClientImpl::GetLayout(GetLayoutOp& inOp, const FileAttr* inAttrPtr)
 {
     inOp.chunks.clear();
+    if (inAttrPtr && 0 == inAttrPtr->numReplicas) {
+        ChunkLayoutInfo chunk;
+        GetAllocOp      op(0, inOp.fid, 0);
+        op.objectStoreFlag = true;
+        inOp.chunks.reserve(
+            (size_t)max(int64_t(0), inAttrPtr->fileSize / (int64_t)CHUNKSIZE));
```

Comment:
The code would'n compile without it, as max template has both parameters to be of the same type.